src/main/java/org/folio/services/locationunit/LibraryService.java

```diff
@@ -0,0 +1,116 @@
+package org.folio.services.locationunit;
+
+import static io.vertx.core.Future.succeededFuture;
+import static org.folio.rest.impl.LocationUnitApi.URL_PREFIX;
+import static org.folio.rest.tools.utils.ValidationHelper.createValidationErrorMessage;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+import java.util.Map;
+import java.util.function.Function;
+import javax.ws.rs.core.Response;
+import org.folio.persist.LibraryRepository;
+import org.folio.rest.jaxrs.model.Loclib;
+import org.folio.rest.jaxrs.model.Loclibs;
+import org.folio.rest.jaxrs.resource.LocationUnits.DeleteLocationUnitsLibrariesByIdResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.DeleteLocationUnitsLibrariesResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.GetLocationUnitsLibrariesByIdResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.GetLocationUnitsLibrariesResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.PostLocationUnitsLibrariesResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.PutLocationUnitsLibrariesByIdResponse;
+import org.folio.rest.persist.PgUtil;
+import org.folio.services.domainevent.LibraryDomainEventPublisher;
+
+public class LibraryService {
+
+  public static final String LIBRARY_TABLE = "loclibrary";
+  private static final String MSG_ID_NOT_MATCH =
+    "Illegal operation: Library ID cannot be changed";
+  private final Context context;
+  private final Map<String, String> okapiHeaders;
+  private final LibraryRepository repository;
+  private final LibraryDomainEventPublisher domainEventService;
+
+  public LibraryService(Context context, Map<String, String> okapiHeaders) {
+    this.context = context;
+    this.okapiHeaders = okapiHeaders;
+    this.repository = new LibraryRepository(context, okapiHeaders);
+    this.domainEventService =
+      new LibraryDomainEventPublisher(context, okapiHeaders);
+  }
+
+  public Future<Response> getByQuery(String cql, int offset, int limit) {
+    return PgUtil.get(LIBRARY_TABLE, Loclib.class, Loclibs.class,
+      cql, offset, limit, okapiHeaders, context,
+      GetLocationUnitsLibrariesResponse.class);
+  }
+
+  public Future<Response> getById(String id) {
+    return PgUtil.getById(LIBRARY_TABLE, Loclib.class, id, okapiHeaders, context,
+      GetLocationUnitsLibrariesByIdResponse.class);
+  }
+
+  public Future<Response> create(Loclib loclib) {
+    return PgUtil.post(LIBRARY_TABLE, loclib, okapiHeaders, context,
+        PostLocationUnitsLibrariesResponse.class)
+      .onSuccess(response ->
+        PostLocationUnitsLibrariesResponse
+          .respond201WithApplicationJson(response,
+            PostLocationUnitsLibrariesResponse.headersFor201()
+              .withLocation(URL_PREFIX + response)))
```

Comment:
should we invoke domainEventService.publishCreated() here?

Reply:
Good catch Thanks, fixed